### ISSUE TITLE
Separate and categorize tests

### ISSUE THREAD

**Issue by carldlaird** [grid-parity-exchange/Egret#33]
- We need to split up the tests so that different test cases are recognized as different tests in pytest. One potential package is the parameterized package.
- We can also categorize tests to avoid long running tests on travis (but still make it easy to run them locally).

**Comment by jwatsonnm** [comment]
bknueven and I discussed introducing more comprehensive unit commitment tests in the very near future, as far more cases are available than what are currently in the repository. However, this would of course blow up the run time. We don't want to run these exhaustive tests on travis all the time, but it would be nice to somehow flag some PRs as requiring them, e.g., for more disruptive changes. I have no idea if this fantasy can be realized, however, through a github/travis combination.

**Comment by michaelbynum** [comment]
@jwatsonnm I agree that this would be ideal. I would guess that it is possible, but it would take time (at least for me) to figure out how.
